api/filters/platform_termination_filter.go

```diff
@@ -62,11 +65,118 @@ func (f *platformTerminationFilter) Run(req *web.Request, next web.Handler) (*we
 				StatusCode:  http.StatusUnprocessableEntity,
 			}
 		}
-	}
+		instanceInOtherPlatforms, err := hasSharedInstancesInOtherPlatforms(ctx, platform, f.repository)
+		if err != nil {
+			return nil, err
+		}
 
+		if instanceInOtherPlatforms {
+			sharedInstancesReferences, err := findReferencesOfSharedInstancesInOtherPlatforms(ctx, platform, f.repository)
+			if err != nil {
+				return nil, err
+			}
+			return nil, &util.HTTPError{
+				ErrorType:   "UnprocessableEntity",
+				Description: "Platform cannot be deleted because other platform(s) has reference instance(s) to the shared instances in given platform. Details: " + formatSharingReferences(sharedInstancesReferences),
+				StatusCode:  http.StatusUnprocessableEntity,
+			}
+		}
+	}
 	return next.Handle(req)
 }
 
+func formatSharingReferences(references []*SharingReferences) string {
+	var msg []string
+	for i := 0; i < len(references); i++ {
+		msg = append(msg, "shared instance "+references[i].sharedInstanceID+" referenced by instance(s) "+strings.Join(references[i].referenceIDs, ", "))
+	}
+	return strings.Join(msg, ", ")
+}
+
+type SharingReferences struct {
+	sharedInstanceID string
+	referenceIDs     []string
+}
+
+func hasSharedInstancesInOtherPlatforms(ctx context.Context, platform *types.Platform, repository storage.Repository) (bool, error) {
```

Comment:
why is this function needed? isn't it enough to get the references and check if len==0? if there are no refs the query is as quick as count, and if there are we get them in a single query instead of 2.

Reply:
currently we query to get the referenced instance details only in case of an error and other wise just proceed the platform cascade delete. for the positive flow it's just the count query.

do you think we should reduce it to list only?

Reply:
I think that when the count is 0 the list query does nothing and doesn't consume any more resources (same index that is used for count is used to understand that there are no results), and if it isn't 0 we already get the data we need without any additional queries. I would drop the count and only work with list.

Reply:
dropped the count